src/objectExplorer/nodes/treeNodeInfo.ts

```diff
@@ -47,26 +51,29 @@ export class TreeNodeInfo extends vscode.TreeItem implements ITreeNodeInfo {
         this._nodeType = nodeType;
         this._sessionId = sessionId;
         this._parentNode = parentNode;
-        this._connectionInfo = connectionInfo;
+        this._connectionProfile = connectionProfile;
         this._filterableProperties = filterProperties;
         this._metadata = objectMetadata;
         this._filters = filters;
-        this.iconPath = ObjectExplorerUtils.iconPath(this.nodeType);
-        if (this.connectionInfo?.database) {
-            if (this.nodeType === Constants.serverLabel) {
-                this.iconPath = ObjectExplorerUtils.iconPath(Constants.database_green);
-            }
-            if (this.nodeType === Constants.disconnectedServerNodeType) {
-                this.iconPath = ObjectExplorerUtils.iconPath(Constants.database_red);
-            }
+        this._nodeSubType = nodeSubType;
+        if (this._nodeSubType) {
+            this.iconPath = ObjectExplorerUtils.iconPath(`${this._nodeType}_${this._nodeSubType}`);
+        } else {
+            this.iconPath = ObjectExplorerUtils.iconPath(this.nodeType);
         }
+        this.id = this.generateId();
+    }
+
+    // Gernating a unique ID for the node
+    protected generateId(): string {
+        return `${this._connectionProfile.id}-${this._nodePath}-${Date.now()}`;
```

Comment:
What does nodePath look like for servers, databases, and databases-under-servers?  Just curious.

Reply:
1. For Server connections:
`<server-name>`
1. For database connections:
`<server-name>/<database-name>`
1. For databases under servers:
`<server-name>/Databases/<database-name>`